crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -63,94 +67,129 @@ def check_all_repos
         repochecks
       end
 
-      def admin_platform
-        NodeObject.admin_node.target_platform
-      end
+      def provided_and_enabled?(feature, platform = nil)
+        answer = false
 
-      def web_port
-        Proposal.where(barclamp: "provisioner").first.raw_attributes["web_port"]
-      end
+        if platform.nil?
+          all_platforms.each do |platform|
+            if provided_and_enabled?(feature, platform)
+              answer = true
+              break
+            end
+          end
+        else
+          found = false
+          answer = true
 
-      def repositories(platform)
-        registry[platform]["repos"].keys
+          repositories(platform).each do |repo|
+            provided_features = registry[platform]["repos"][repo]["features"] || []
+            if provided_features.include? feature
```

Comment:
Style/Next: Use `next` to skip iteration. (https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)